src/rateEngine/RateElement.ts

```diff
@@ -21,24 +16,32 @@ class RateElement {
   billingCategory?: BillingCategory;
   errors: Array<ValidatorError> = [];
 
-  constructor(rateElementArgs: RateElementInterface, loadProfile: LoadProfile, otherRateElements: Array<RateElementInterface> = []) {
+  constructor(
+    rateElementArgs: RateElementInterface,
+    loadProfile: LoadProfile,
+    otherRateElements: Array<RateElementInterface> = [],
+  ) {
     const { id, rateElementType, name, billingCategory } = rateElementArgs;
     this.id = id;
     this.name = name;
     this.type = rateElementType;
     this.billingCategory = billingCategory;
 
     if (RateCalculator.shouldValidate) {
-      const validator = ValidatorFactory.make(rateElementType, rateElementArgs['rateComponents'], loadProfile).validate();
-      RateCalculator.shouldLogValidationErrors && validator.reportErrors();
+      const validator = ValidatorFactory.make(
+        rateElementType,
+        rateElementArgs['rateComponents'] ?? [],
```

Comment:
why wouldn't we destructure this in line 24 like the rest of the `rateElementArgs` properties?

Reply:
Not sure, my goal here was to leave as much untouched as possible. The only change in this file is formatting and null/undefined checking with `??`.

Reply:
fair enough... i wonder why it was written that way